### PR TITLE
Flipping CI to use Travis

### PR DESCRIPTION
**Pull request by rjlohan**
*Issue #, if available:* #4 

*Description of changes:* Converts the CI build to use Travis and also adds pre-commit and applies to the current repo.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+dist: bionic
+jdk: openjdk11
+env:
+  global:
+    - PYENV_VERSION=3.7
+install:
+  - pip3 install --user pre-commit
+script:
+  - pre-commit run --all-files --verbose
+  - mvn clean verify
```
Comment (PatMyron) [inline .travis.yml:11]: @rjlohan what's the difference between this and the previous `mvn compile test`?

Comment (rjlohan) [reply .travis.yml:11]: Practically, nothing in this repo. But `verify` would also check code coverage requirements and such if they were in the `pom.xml`

Comment (rjlohan) [reply .travis.yml:11]: Build lifecycle reference: https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html

Comment (PatMyron) [reply .travis.yml:11]: @rjlohan thoughts on adding something like this to test it working end-to-end?
```bash
mvn clean package
echo 'settings:
  regions: [us-east-1]
  output: cfn-schemas' > cfg.yml
java -jar target/aws-cloudformation-template-schema-1.0-SNAPSHOT-jar-with-dependencies.jar --config-file cfg.yml
```

**Pull request merged**